src/Bases/ModelBase.py

```diff
@@ -99,25 +99,31 @@ def isDynamic(self):
     """
     return self._dynamicHandling
 
-  def setVariable(self, value):
+  def setVariable(self, name, value):
     """
       Set value if a float/int/list is provided in the node text, otherwise treat the provided value as RAVEN variable
+      @ In, name, str, the name of variable
       @ In, value, str or float or list, the value of given variable
-      @ Out, ret, str or float or numpy.array, the recasted value
+      @ Out, None
     """
     ret = None
     # multi-entry or single-entry?
     if len(value) == 1:
       if not utils.isAFloatOrInt(value[0]):
         ret = value[0]
-      else:
+        ## add it to _variableDict, so the code can aware of this variable
+        if name in self._variableDict.keys():
+          raise KeyError('Duplicate variable: "{}" is found! Please check your input file.'.format(name))
+        self._variableDict[name] = ret
+      else: ## string is provided, so treat it as RAVEN variables
         ret = np.atleast_1d(value)
     else:
       # should be floats; InputData assures the entries are the same type already
       if not utils.isAFloatOrInt(value[0]):
```

Comment:
we should we check all entries, not only the first one. no?

Reply:
We do the conversion in the InputType. If first entry is str, we will convert the rest to str, if first entry is float, we will convert the rest to float: 
![image](https://user-images.githubusercontent.com/7321071/121390405-931ded00-c90a-11eb-9186-dc7291fa263d.png)

Reply:
gotcha, thanks